src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderChunk_Common.java

```diff
@@ -0,0 +1,140 @@
+ * <p>
+ * Allows to render cubes outside of 0..256 height range.
+ */
+@MethodsReturnNonnullByDefault
+@ParametersAreNonnullByDefault
+@Mixin(RenderChunk.class)
+public abstract class MixinRenderChunk_Common implements IRenderChunk {
+    
+    @Shadow @Final public World world;
+    @Shadow @Final private BlockPos.MutableBlockPos position;
+    
+    /** Warning! Field mixed across mixins: 
+     * {@link cubicchunks.asm.mixin.selectable.client.MixinRenderChunk_OptifineSpecific} and this. */
+    private Cube[] cubeCache;
+    /** Warning! Field mixed across mixins: 
+     * {@link cubicchunks.asm.mixin.selectable.client.MixinRenderChunk_OptifineSpecific} and this. */
+    private Chunk[] chunkCache;
+    
+    @ModifyConstant(method = "setPosition", constant = @Constant(intValue = 16))
+    public int onSetPosition(int oldValue) {
+        return RenderVariables.getRenderChunkSize();
+    }
+    
+    @ModifyConstant(method = "rebuildChunk", constant = @Constant(intValue = 15))
+    public int onRebuildChunk(int oldValue) {
+        return RenderVariables.getRenderChunkMaxPos();
+    }
+    
+    @ModifyConstant(method = "getDistanceSq", constant = @Constant(doubleValue = 8.0D))
+    public double onDistanceSq(double oldValue) {
+        return RenderVariables.getRenderChunkCenterPos();
+    }
+    
+    @Override
+    public boolean hasEntities() {
+        boolean hasEntities = false;
+        ICubicWorldClient cworld = (ICubicWorldClient) world;
+        int renderChunkCubeSize = RenderVariables.getRenderChunkSize() / Cube.SIZE;
+        if (cworld.isCubicWorld()) {
+            if (cubeCache == null) {
+                cubeCache = new Cube[1 << RenderVariables.getRenderChunkPosShitBit() * 3];
+                CubeProviderClient cubeProvider = cworld.getCubeCache();
+                int cx0 = Coords.blockToCube(position.getX());
+                int cy0 = Coords.blockToCube(position.getY());
+                int cz0 = Coords.blockToCube(position.getZ());
+                int index = 0;
+                for (int cx = cx0; cx < cx0 + renderChunkCubeSize; cx++)
+                    for (int cy = cy0; cy < cy0 + renderChunkCubeSize; cy++)
+                        for (int cz = cz0; cz < cz0 + renderChunkCubeSize; cz++) {
+                            Cube cube = cubeProvider.getCube(cx, cy, cz);
+                            cubeCache[index++] = cube;
+                            if (!hasEntities && cube.getEntityContainer().size() != 0)
+                                hasEntities = true;
+                        }
+                return hasEntities;
+            }
+            for (Cube cube : cubeCache) {
+                if (cube.getEntityContainer().size() != 0) {
+                    return true;
+                }
+            }
+        } else {
+            int y0 = position.getY();
+            if (y0 < 0 || y0 >= world.getHeight()) {
+                return false;
+            }
+            int cy0 = Coords.blockToCube(y0);
```

Comment:
Rename the variables

Reply:
To what? `chunkPosY` and `chunkPosStartY`?

Reply:
If that's what it does - yes. Something more readable than `y0` and `cy0`